### PR TITLE
Build: Bump go version to 1.20.5

### PR DESCRIPTION
**Pull request by sethiyash**
Bumping Golang version to 1.20.5 to get rid of the CVEs present in Go runtime less than 1.20.5

### CODE REVIEW

**Comment by sethiyash** [comment]
Can I get approval for these changes? These are critical for us to get rid of the CVE's in go runtime.
cc: @autrilla @jvehent @ajvb @oremj @adrianosela

File: Dockerfile.alpine [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.18-alpine3.16 AS builder
```
Comment (adrianosela) [inline Dockerfile.alpine:1]: latest alpine is 3.18

**Review by adrianosela — APPROVED** [review]
Change LGTM -- but unfortunately I do not have approval/merge privileges for this repo :)

**Comment by felixfontein** [comment]
> Can I get approval for these changes? These are critical for us to get rid of the CVE's in go runtime.

Please look at https://github.com/mozilla/sops/discussions/927#discussioncomment-6275964.

**Comment by hiddeco** [comment]
Addressed by #1148

**Pull request closed without merging**